File: src/Implementations/Beer.cpp
#include "Beer.h"
#include "GameState.h"

void Beer::UseItem(GameState& gameState)
{
	if (gameState.magazine.CheckBulletType())
	{
		gameState.magazine.DecreaseFullCount();
		if (gameState.gameStateManager.GetStarter() == GameEnums::STARTER_HUMAN)
		{
			gameState.log.push_back("Uzyles piwa (usunieto pelna)!");
		}
		else
		{
			gameState.log.push_back("Komputer uzyl piwa (usunieto pelna)!");
		}
	}
	else
	{
		gameState.magazine.DecreaseEmptyCount();
		if (gameState.gameStateManager.GetStarter() == GameEnums::STARTER_HUMAN)
		{
			gameState.log.push_back("Uzyles piwa (usunieto pusta)!");
		}
		else
		{
			gameState.log.push_back("Komputer uzyl piwa (usunieto pusta)!");
		}
	}
	gameState.magazine.DecreaseBulletCount();
}

File: C#/src/Implementations/Beer.cs
public class Beer : Item
{
    public Beer() { }

    public void UseItem(GameState gameState)
    {
        if (gameState.magazine.CheckBulletType())
        {
            gameState.magazine.DecreaseFullCount();
            if (gameState.gameStateManager.GetStarter() == (int)GameEnums.Starter.STARTER_HUMAN)
            {
                gameState.log.Add("Uzyles piwa (usunieto pelna)!");
            }
            else
            {
                gameState.log.Add("Komputer uzyl piwa (usunieto pelna)!");
            }
        }
        else
        {
            gameState.magazine.DecreaseEmptyCount();
            if (gameState.gameStateManager.GetStarter() == (int)GameEnums.Starter.STARTER_HUMAN)
            {
                gameState.log.Add("Uzyles piwa (usunieto pusta)!");
            }
            else
            {
                gameState.log.Add("Komputer uzyl piwa (usunieto pusta)!");
            }
        }
        gameState.magazine.DecreaseBulletCount();
    }
}